site/app/entities/plagiarism/PlagiarismConfig.php

```diff
@@ -0,0 +1,336 @@
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @var string
+     */
+    protected $gradeable_id;
+
+    /**
+     * @ORM\Column(type="smallint")
+     * @var int
+     */
+    protected $config_id;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @var string
+     */
+    protected $version;
+
+    /**
+     * @ORM\Column(type="simple_array")
+     * @var array
+     */
+    protected $regex;
+
+    /**
+     * @ORM\Column(type="boolean")
+     * @var bool
+     */
+    protected $regex_dir_submissions;
+
+    /**
+     * @ORM\Column(type="boolean")
+     * @var bool
+     */
+    protected $regex_dir_results;
+
+    /**
+     * @ORM\Column(type="boolean")
+     * @var bool
+     */
+    protected $regex_dir_checkout;
+
+    /**
+     * @ORM\Column(type="string", length=255)
+     * @var string
+     */
+    protected $language;
+
+    /**
+     * @ORM\Column(type="smallint")
+     * @var int
+     */
+    protected $threshold;
+
+    /**
+     * @ORM\Column(type="smallint")
+     * @var int
+     */
+    protected $sequence_length;
+
+    /**
+     * @ORM\Column(type="json")
+     * @var array
+     */
+    protected $other_gradeables;
+
+    /**
+     * @ORM\Column(type="simple_array")
+     * @var array
+     */
+    protected $ignore_submissions;
+
+    /* FUNCTIONS */
+
+    /**
+     * PlagiarismConfig constructor.
+     * @param string $gradeable_id
+     * @param int $config_id
+     * @param string $version
+     * @param array $regex
+     * @param bool $regex_dir_submissions
+     * @param bool $regex_dir_results
+     * @param bool $regex_dir_checkout
+     * @param string $language
+     * @param int $threshold
+     * @param int $sequence_length
+     * @param array $other_gradeables
+     * @param array $ignored_submissions
+     * @throws Exception
+     */
+    public function __construct(
+        string $gradeable_id,
+        int $config_id,
+        string $version,
+        array $regex,
+        bool $regex_dir_submissions,
+        bool $regex_dir_results,
+        bool $regex_dir_checkout,
+        string $language,
+        int $threshold,
+        int $sequence_length,
+        array $other_gradeables,
+        array $ignored_submissions
+    ) {
+        $this->gradeable_id = $gradeable_id;
+        $this->config_id = $config_id;
+        $this->setVersionStatus($version);
+        $this->setRegexArray($regex);
+        $this->setRegexDirSubmissions($regex_dir_submissions);
+        $this->setRegexDirResults($regex_dir_results);
+        $this->setRegexDirCheckout($regex_dir_checkout);
+        $this->setLanguage($language);
+        $this->setThreshold($threshold);
+        $this->setSequenceLength($sequence_length);
+        $this->setOtherGradeables($other_gradeables);
+        $this->setIgnoredSubmissions($ignored_submissions);
+    }
+
+    /**
+     * @return int
+     */
```

Comment:
These docstrings are not useful and are already covered by the type hint on the function. Unless there's meaningful documentation within the comment, there's (in my opinion) no reason to have it as it just means there's two places instead of one we need to update if we were to change the variable's type for some reason.